synapse/storage/events.py

```diff
@@ -2191,6 +2191,89 @@ def _find_unreferenced_groups_during_purge(self, txn, state_groups):
 
         return to_delete, to_dedelta
 
+    def purge_room(self, room_id):
+        """Deletes all record of a room
+
+        Args:
+            room_id (str):
+        """
+
+        return self.runInteraction("purge_room", self._purge_room_txn, room_id)
+
+    def _purge_room_txn(self, txn, room_id):
+        # first we have to delete the state groups states
+        logger.info("[purge] removing %s from state_groups_state", room_id)
+
+        txn.execute(
+            "DELETE FROM state_groups_state "
+            "WHERE state_group IN ("
+            "SELECT state_group FROM events JOIN event_to_state_groups USING(event_id) "
+            "WHERE events.room_id=?"
+            ")",
```

Comment:
I really do prefer using multiline strings for this sort of stuff, as it makes it a *lot* easier to then c+p to debug.

Reply:
fair, yes. For some reason I forget to do this.